File: 백준/Bronze/11943. 파일 옮기기/파일 옮기기.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        StringTokenizer st = new StringTokenizer(br.readLine());
        // 첫 번째 바구니: 사과 a개, 오렌지 b개
        int a = Integer.parseInt(st.nextToken());
        int b = Integer.parseInt(st.nextToken());

        st = new StringTokenizer(br.readLine());
        // 두 번째 바구니: 사과 c개, 오렌지 d개
        int c = Integer.parseInt(st.nextToken());
        int d = Integer.parseInt(st.nextToken());

        // 경우 1: 바구니 1을 사과만, 바구니 2를 오렌지만
        // b개의 오렌지를 바구니 2로 + c개의 사과를 바구니 1로
        int case1 = b + c;

        // 경우 2: 바구니1을 오렌지만, 바구니 2를 사과만
        // a개의 사과를 바구니 2로 + d개의 오렌지를 바구니 1로
        int case2 = a + d;

        // 두 경우 중 최솟값 출력
        bw.write(String.valueOf(Math.min(case1, case2)));
        
        bw.flush();
        bw.close();
        br.close();
    }
}

File: 백준/Bronze/11943. 파일 옮기기/파일 옮기기.py
import sys

a, b = map(int, sys.stdin.readline().split())
c, d = map(int, sys.stdin.readline().split())

# 첫 번째 바구니 사과:a, 오렌지:b
# 두 번째 바구니 사과:c, 오렌지:d
# 첫 번째 바구니는 사과만, 두 번째 바구니는 오렌지만 있게 하는 경우
move_1 = b+c
# 첫 번째 바구니는 오렌지만, 두 번째 바구니는 사과만 있게 하는 경우
move_2 = a+d
ans = min(move_1, move_2)
sys.stdout.write(str(ans))
